src/clients/aggregator_client/api_types.rs

```diff
@@ -153,44 +153,45 @@ impl From<AggregatorVdaf> for Vdaf {
 #[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq, Eq)]
 pub enum QueryType {
     TimeInterval,
-    FixedSize { max_batch_size: u64 },
+    FixedSize {
+        max_batch_size: u64,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        batch_time_window_size: Option<u64>,
+    },
 }
 
-impl QueryType {
-    pub fn name(&self) -> QueryTypeName {
-        match self {
-            QueryType::TimeInterval => QueryTypeName::TimeInterval,
-            QueryType::FixedSize { .. } => QueryTypeName::FixedSize,
+impl From<ProvisionableTask> for QueryType {
+    fn from(value: ProvisionableTask) -> Self {
+        if let Some(max_batch_size) = value.max_batch_size {
+            Self::FixedSize {
+                max_batch_size,
+                batch_time_window_size: value.batch_time_window_size_seconds,
+            }
+        } else {
+            Self::TimeInterval
         }
     }
 }
 
-impl From<QueryType> for Option<i64> {
-    fn from(value: QueryType) -> Self {
-        Option::<u64>::from(value).map(|u| u.try_into().unwrap())
-    }
-}
-
-impl From<QueryType> for Option<u64> {
-    fn from(value: QueryType) -> Self {
-        match value {
-            QueryType::TimeInterval => None,
-            QueryType::FixedSize { max_batch_size } => Some(max_batch_size),
+impl From<NewTask> for QueryType {
```

Comment:
I think this would be better suited as a new method on `NewTask` that takes `&self`, rather than a `From` conversion. Each place we use it, we clone the entire task first.